vaadin-text-field.html

```diff
@@ -347,6 +347,12 @@
           this._setHasValue(newVal !== '' && newVal != null);
         }
 
+        _clickHandler(e) {
+          if (e.target.hasAttribute('prevent-focus')) {
```

Comment:
I think this won’t catch cases when there are nested elements inside the prefix/suffix element, for example: 

```html
<vaadin-text-field>
  <vaadin-button prevent-focus>
    <iron-icon icon="plus"></iron-icon>
  </vaadin-button>
</vaadin-text-field>
```

The target would most probably be the `iron-icon` without the attribute.